src/compositions/usePrefetchedProps.ts

```diff
@@ -0,0 +1,30 @@
+import { MaybeRef, Ref, ref, watch } from 'vue'
+import { usePropStore } from '@/compositions/usePropStore'
+import { PrefetchConfigs } from '@/types/prefetch'
+import { ResolvedRoute } from '@/types/resolved'
+
+export type UsePrefetchedProps = {
+  prefetchProps: Ref<Record<string, unknown>>,
+  commitPrefetchedProps: () => void,
+}
+
+export function usePrefetchedProps(route: MaybeRef<ResolvedRoute | undefined>, prefetch: MaybeRef<PrefetchConfigs | undefined>): UsePrefetchedProps {
```

Comment:
Another spot to use `MaybeRefOrGetter`

One thing I did in my version of this was may this a single object so you could do this

```
usePrefetchedProps(() => ({
  route: route.value,
  routerPrefetch: router.prefetch,
  linkPrefetch: options.prefetch
}))
```

Don't feel strongly, just felt nice when I was playing with it.